backend/common/src/test/java/ai/verta/modeldb/common/futures/FutureJdbiTest.java

```diff
@@ -0,0 +1,82 @@
+package ai.verta.modeldb.common.futures;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import io.grpc.Context;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jdbi.v3.core.statement.StatementExceptions;
+import org.junit.jupiter.api.Test;
+
+class FutureJdbiTest {
+
+  @Test
+  void contextWithHandle() throws Exception {
+    InternalJdbi internalJdbi = createMockInternalJdbi(mock(Handle.class));
+
+    FutureJdbi subject = new FutureJdbi(internalJdbi, FutureExecutor.newSingleThreadExecutor());
+
+    Context.Key<String> testKey = Context.key("cheese");
+
+    Context contextWithData = Context.ROOT.withValue(testKey, "cheddar");
+    AtomicReference<String> captured = new AtomicReference<>();
+    InternalFuture<String> result =
+        contextWithData
+            .wrap(
+                () ->
+                    subject.withHandle(
+                        h -> {
+                          captured.set(testKey.get());
+                          return "foo";
+                        }))
+            .call();
```

Comment:
nit: resolve the future here. Otherwise it's easy to flip the two asserts below and get a failure

Reply:
fixed.